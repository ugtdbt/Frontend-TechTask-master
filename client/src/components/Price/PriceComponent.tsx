import React, { memo } from "react";
import type { FC } from "react";
import Typography from "@mui/material/Typography";
import { formatter } from "../../utils/formatter";

interface PriceProps {
  price: number;
}

const Price: FC<PriceProps> = memo((props: PriceProps) => {
  const { price } = props;

  return (
    <Typography gutterBottom variant="h5" component="div" data-testid="price">
      {formatter.format(price / 100)}
    </Typography>
  );
});

Price.displayName = "Price";
export default Price;
