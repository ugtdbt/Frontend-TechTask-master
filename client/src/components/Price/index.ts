export { default as Price } from "./PriceComponent";
