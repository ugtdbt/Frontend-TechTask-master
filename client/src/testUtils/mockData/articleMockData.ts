export const article = {
  imageUrl:
    "https://cdn1.home24.net/images/media/catalog/product/200x200/png/l/a/lattenrost-super-fix-90-x-200cm-4448116.webp",
  name: "Lattenrost Super Fix",
  variantName: "Breite: 140 cm",
  price: 14990,
};
