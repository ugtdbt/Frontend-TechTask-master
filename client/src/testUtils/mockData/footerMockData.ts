export const footer = {
  description:
    "Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten.",
};
