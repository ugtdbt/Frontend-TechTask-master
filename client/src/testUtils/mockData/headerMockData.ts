export const sections = [
  { title: "furniture", url: "#" },
  { title: "lamps", url: "#" },
  { title: "accessories", url: "#" },
  { title: "textiles", url: "#" },
  { title: "children", url: "#" },
  { title: "kitchen", url: "#" },
  { title: "bath", url: "#" },
  { title: "Garden", url: "#" },
  { title: "Brands", url: "#" },
  { title: "Shop the look", url: "#" },
  { title: "advisory", url: "#" },
  { title: "stores", url: "#" },
  { title: "Sale", url: "#" },
];

export const header = {
  sections,
  title: "Home24",
};
