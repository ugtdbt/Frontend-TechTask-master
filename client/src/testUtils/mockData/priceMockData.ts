export const price = {
  price: 2500,
};
