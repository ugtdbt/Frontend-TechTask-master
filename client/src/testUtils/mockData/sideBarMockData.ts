export const sideBar = {
  list: [
    { name: "Wohnzimmer", urlPath: "wohnzimmer-moebel/" },
    { name: "Polsterm√∂bel", urlPath: "alle-polstermoebel/" },
  ],
};
