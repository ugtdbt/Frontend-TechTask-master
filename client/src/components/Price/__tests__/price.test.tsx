import { cleanup, render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import { price } from "../../../testUtils/mockData/priceMockData";
import Price from "../PriceComponent";

describe("Price component", () => {
  const PriceComponent = <Price price={price.price} />;

  afterEach(() => {
    cleanup();
  });

  it("Price match Snapshot", () => {
    const tree = renderer.create(PriceComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render price component", () => {
    render(PriceComponent);
    const priceElem = screen.getByTestId("price");
    expect(priceElem).toBeInTheDocument();
  });

  it("Check price text", () => {
    render(PriceComponent);
    const priceElem = screen.getByTestId("price")?.innerHTML;
    expect(priceElem).toContain("25,00&nbsp;â‚¬");
  });
});
