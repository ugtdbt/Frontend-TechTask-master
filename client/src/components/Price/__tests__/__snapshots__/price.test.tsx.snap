// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component Price match Snapshot 1`] = `
<div
  className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
  data-testid="price"
>
  25,00 €
</div>
`;
